src/control/cmd/dmg/system.go

```diff
@@ -259,6 +260,52 @@ func (cmd *systemStopCmd) Execute(_ []string) error {
 	return displaySystemAction(cmd.log, resp.Results)
 }
 
+// systemReformatCmd is the struct representing the command to start system.
+type systemReformatCmd struct {
+	logCmd
+	ctlClientCmd
+	jsonOutputCmd
+	Verbose bool   `short:"v" long:"verbose" description:"Show results of each SCM & NVMe device format operation"`
+	Ranks   string `long:"ranks" short:"r" description:"Comma separated list of system ranks to query"`
+}
+
+// Execute is run when systemReformatCmd activates
+func (cmd *systemReformatCmd) Execute(_ []string) error {
+	var ranks []uint32
+	if err := common.ParseNumberList(cmd.Ranks, &ranks); err != nil {
+		return errors.Wrap(err, "parsing input ranklist")
+	}
+
+	ctx := context.Background()
+	resp, err := control.SystemReformat(ctx, cmd.ctlClient, &control.SystemReformatReq{
+		Ranks: system.RanksFromUint32(ranks),
+	})
+	if err != nil {
+		return errors.Wrap(err, "System-Reformat command failed")
+	}
+
+	if cmd.jsonOutputEnabled() {
+		return cmd.outputJSON(os.Stdout, resp)
+	}
+
+	if len(resp.Results) != 0 {
+		cmd.log.Debug("System-Reformat failed to put ranks in 'await format' state")
+		return displaySystemAction(cmd.log, resp.Results)
+	}
```

Comment:
intention here is to avoid reformat if ranks cannot be set into "await format state" and it won't proceed to try to format storage in which case the host errors won't be relevant. if hosts do error in the "setting await format" phase, results will be generated for unresponsive hosts regardless.
Alternatively I could remove any ranks that failed the "await format state", populating the HostErrors as you say, then continue with the next phase with a filtered rank list that doesn't contain the failed rank. is that what you would like?

Reply:
Yeah, I realized that this was the intent after I started looking at the implementation of `control.SystemFormat()`, hence my objection to having different parts of the response struct mean different things depending on how things went.

No, I don't necessarily think it makes sense to try to format if some ranks failed to get into the correct state. What I'd like to see is that the logic in `control.SystemFormat()` takes care of interpreting the success/failure of the prep stage and returns a response struct that is consistent with other response structs in the API (i.e. populate HostErrors with failures).